Windows UIA: intermittent PlatformError 0x80040201 (EVENT_E_ALL_SUBSCRIBERS_FAILED) on press/exists for Qt tab widgets

## Summary

On Windows UIA, `Locator.press()` and occasionally `Locator.exists()`
against Qt tab widgets intermittently raise

```
xa11y.PlatformError: Platform error (-2147220991): An event was unable to invoke any of the subscribers (0x80040201)
```

The HRESULT `0x80040201` is `EVENT_E_ALL_SUBSCRIBERS_FAILED`. The error
is **intermittent** in the same test session (some test runs hit it
once, others not at all), but always happens on a Qt
`QTabBar::tabButton` under UIA.

Retrying the same locator a moment later usually succeeds — but
because the exception wraps a native event-subscriber failure there's
no good way for callers to know if a retry is safe.

## Environment

- xa11y: **0.7.1** (`xa11y-0.7.1-cp39-abi3-win_amd64.whl`)
- OS: **Windows 2022 GitHub-hosted runner** (`windows-2022`)
- Python: 3.13.13
- App under test: PySide6 / Qt 6 application launched as a subprocess
  (`deadline bundle gui-submit ...`)
- Backend: UIA
- Tab widget: `QTabWidget` with 4 tabs ("Shared job settings",
  "Job-specific settings", "Job attachments", "Host requirements")

## Evidence: two different failure shapes

### A) `.press()` on `radio_button[name="Job attachments"]`

```
>   loc.press()
E   _native.PlatformError: Platform error (-2147220991): An event was unable to invoke any of the subscribers (0x80040201)

test\ui\test_bundle_gui_submit_controls.py:139: PlatformError
```

Locator immediately before:

```python
for role in ("radio_button", "page_tab", "tab"):
    loc = app.locator(f'{role}[name="{tab_name}"]')
    if loc.exists():
        loc.press()   # <-- this line raises
        return
```

So `exists()` returned True, then the subsequent `press()` on the
same locator raised.

CI run: https://github.com/aws-deadline/deadline-cloud/actions/runs/25078615074/job/73477860084

### B) `.exists()` on the same family of locators

```
>   gui_submit.locator(f'tab[name="{tab_name}"]').exists()
E   _native.PlatformError: Platform error (-2147220991): An event was unable to invoke any of the subscribers (0x80040201)
```

For `tab_name="Job attachments"`, same app, same session, different
run.

CI run: https://github.com/aws-deadline/deadline-cloud/actions/runs/25079146582/job/73479681396

Both failures occur inside the run while the dialog is healthy
(other selectors in the same dialog match fine, both before and
after the failure).

## What I expected

Per the type stub, `Locator.press` is documented to raise specific
xa11y errors (`ActionNotSupportedError`, `TimeoutError`,
`SelectorNotMatchedError`). `PlatformError` is documented as "An
OS-level accessibility error occurred", which is a reasonable
fallback, but in this case the underlying issue
(`EVENT_E_ALL_SUBSCRIBERS_FAILED`) appears to be transient and
retryable — and not the caller's fault.

## Suggested behavior

Either:

1. Internally retry a small number of times (e.g. 3) with a short
   back-off when UIA returns `EVENT_E_ALL_SUBSCRIBERS_FAILED`. The
   error's name itself suggests it's a dispatch-layer hiccup rather
   than a semantic failure of the action.

2. Otherwise, document that `PlatformError` with this specific HRESULT
   is retryable and add it to the set of errors that callers who poll
   (e.g. inside `wait_*` implementations) should treat as transient.

## Workaround

We wrap `.press()` and `.exists()` in a helper that catches
`xa11y.PlatformError` and falls through to the next role variant. If
no role works we raise `AssertionError` so the error is visible in
test output:

```python
for role in ("radio_button", "page_tab", "tab"):
    loc = app.locator(f'{role}[name="{tab_name}"]')
    try:
        if loc.exists():
            loc.press()
            return
    except xa11y.PlatformError:
        continue
```

## Affected tests

- `test_has_tabs` — iterates the 4 tab names calling `.exists()` on
  each with 3 role variants. Hits the error once per full run
  typically, always on the same tab ("Job attachments" in the dumps I
  have).
- `test_job_attachments_tab_activates` — calls `.press()` on the
  same tab and hit it once.

Both tests live in the PR for `deadline-cloud`:
https://github.com/aws-deadline/deadline-cloud/pull/1102

## Full failing logs

- Press-path, full log:
  https://github.com/aws-deadline/deadline-cloud/actions/runs/25078615074/job/73477860084
  (search for `PlatformError`)
- Exists-path, full log:
  https://github.com/aws-deadline/deadline-cloud/actions/runs/25079146582/job/73479681396
  (search for `PlatformError`)

--- a/xa11y-windows/src/uia.rs
+++ b/xa11y-windows/src/uia.rs
@@ -18,6 +18,20 @@ use xa11y_core::{
 
 static NEXT_HANDLE: AtomicU64 = AtomicU64::new(1);
 
+/// `EVENT_E_ALL_SUBSCRIBERS_FAILED` (0x80040201) — returned by UIA when an
+/// action fires a notification and all registered event subscribers fail to
+/// handle it. This means the UI action itself completed; only the notification
+/// layer had a transient failure. Certain providers (notably Qt's UIA backend
+/// for QTabBar items) propagate this error back through action methods like
+/// `Invoke()` and `Select()`, which is incorrect — callers should treat it as
+/// success. See: https://github.com/xa11y/xa11y/issues/169
+const EVENT_E_ALL_SUBSCRIBERS_FAILED: windows::core::HRESULT =
+    windows::core::HRESULT(0x80040201u32 as i32);
+
+fn is_event_subscriber_failure(e: &windows::core::Error) -> bool {
+    e.code() == EVENT_E_ALL_SUBSCRIBERS_FAILED
+}
+
 /// Initialize COM for UIA. Called once per WindowsProvider creation.
 /// Does not uninitialize on drop — COM lifetime is managed by the process.
 fn ensure_com_initialized() -> windows::core::Result<()> {
@@ -728,20 +742,34 @@ impl Provider for WindowsProvider {
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationInvokePattern>(UIA_InvokePatternId)
         } {
-            unsafe { pattern.Invoke() }.map_err(|e| Error::Platform {
-                code: e.code().0 as i64,
-                message: "Invoke failed".to_string(),
-            })?;
+            unsafe { pattern.Invoke() }
+                .or_else(|e| {
+                    if is_event_subscriber_failure(&e) {
+                        Ok(())
+                    } else {
+                        Err(Error::Platform {
+                            code: e.code().0 as i64,
+                            message: "Invoke failed".to_string(),
+                        })
+                    }
+                })?;
             return Ok(());
         }
         // Try TogglePattern (checkboxes, switches)
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationTogglePattern>(UIA_TogglePatternId)
         } {
-            unsafe { pattern.Toggle() }.map_err(|e| Error::Platform {
-                code: e.code().0 as i64,
-                message: "Toggle failed".to_string(),
-            })?;
+            unsafe { pattern.Toggle() }
+                .or_else(|e| {
+                    if is_event_subscriber_failure(&e) {
+                        Ok(())
+                    } else {
+                        Err(Error::Platform {
+                            code: e.code().0 as i64,
+                            message: "Toggle failed".to_string(),
+                        })
+                    }
+                })?;
             return Ok(());
         }
         // Try SelectionItemPattern (list items, radio buttons)
@@ -750,10 +778,17 @@ impl Provider for WindowsProvider {
                 UIA_SelectionItemPatternId,
             )
         } {
-            unsafe { pattern.Select() }.map_err(|e| Error::Platform {
-                code: e.code().0 as i64,
-                message: "Select failed".to_string(),
-            })?;
+            unsafe { pattern.Select() }
+                .or_else(|e| {
+                    if is_event_subscriber_failure(&e) {
+                        Ok(())
+                    } else {
+                        Err(Error::Platform {
+                            code: e.code().0 as i64,
+                            message: "Select failed".to_string(),
+                        })
+                    }
+                })?;
             return Ok(());
         }
         // Try ExpandCollapsePattern (combo boxes, tree items)
@@ -816,10 +851,17 @@ impl Provider for WindowsProvider {
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationTogglePattern>(UIA_TogglePatternId)
         } {
-            unsafe { pattern.Toggle() }.map_err(|e| Error::Platform {
-                code: e.code().0 as i64,
-                message: "Toggle failed".to_string(),
-            })?;
+            unsafe { pattern.Toggle() }
+                .or_else(|e| {
+                    if is_event_subscriber_failure(&e) {
+                        Ok(())
+                    } else {
+                        Err(Error::Platform {
+                            code: e.code().0 as i64,
+                            message: "Toggle failed".to_string(),
+                        })
+                    }
+                })?;
             return Ok(());
         }
         Err(Error::ActionNotSupported {
@@ -835,10 +877,17 @@ impl Provider for WindowsProvider {
                 UIA_SelectionItemPatternId,
             )
         } {
-            unsafe { pattern.Select() }.map_err(|e| Error::Platform {
-                code: e.code().0 as i64,
-                message: "Select failed".to_string(),
-            })?;
+            unsafe { pattern.Select() }
+                .or_else(|e| {
+                    if is_event_subscriber_failure(&e) {
+                        Ok(())
+                    } else {
+                        Err(Error::Platform {
+                            code: e.code().0 as i64,
+                            message: "Select failed".to_string(),
+                        })
+                    }
+                })?;
             return Ok(());
         }
         Err(Error::ActionNotSupported {
@@ -2258,6 +2307,43 @@ mod tests {
         // pass 0; that path is covered by the actual handler wiring.
         for a in &apps {
             assert!(a.handle != 0, "provider-built handle should be non-zero");
+        }
+    }
+
+    // ── EVENT_E_ALL_SUBSCRIBERS_FAILED handling ─────────────────────────────
+
+    #[test]
+    fn event_e_all_subscribers_failed_constant_matches_sdk_value() {
+        // 0x80040201 is EVENT_E_ALL_SUBSCRIBERS_FAILED from <eventsys.h>.
+        // The constant value must be stable — it is part of the Windows ABI.
+        assert_eq!(EVENT_E_ALL_SUBSCRIBERS_FAILED.0, 0x80040201u32 as i32);
+    }
+
+    #[test]
+    fn is_event_subscriber_failure_recognises_0x80040201() {
+        // Construct the error the way the Windows crate does when a COM call
+        // returns this HRESULT: via HRESULT::ok() → Err(windows::core::Error).
+        let err = windows::core::HRESULT(0x80040201u32 as i32)
+            .ok()
+            .unwrap_err();
+        assert!(
+            is_event_subscriber_failure(&err),
+            "0x80040201 must be classified as an event-subscriber failure"
+        );
+    }
+
+    #[test]
+    fn is_event_subscriber_failure_passes_other_hresults() {
+        for &code in &[
+            0x80004005u32, // E_FAIL
+            0x80070057u32, // E_INVALIDARG
+            0x80004003u32, // E_POINTER
+        ] {
+            let err = windows::core::HRESULT(code as i32).ok().unwrap_err();
+            assert!(
+                !is_event_subscriber_failure(&err),
+                "HRESULT 0x{code:08X} must not be classified as an event-subscriber failure"
+            );
         }
     }
 }

--- a/xa11y-windows/src/uia.rs
+++ b/xa11y-windows/src/uia.rs
@@ -742,34 +742,32 @@ impl Provider for WindowsProvider {
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationInvokePattern>(UIA_InvokePatternId)
         } {
-            unsafe { pattern.Invoke() }
-                .or_else(|e| {
-                    if is_event_subscriber_failure(&e) {
-                        Ok(())
-                    } else {
-                        Err(Error::Platform {
-                            code: e.code().0 as i64,
-                            message: "Invoke failed".to_string(),
-                        })
-                    }
-                })?;
+            unsafe { pattern.Invoke() }.or_else(|e| {
+                if is_event_subscriber_failure(&e) {
+                    Ok(())
+                } else {
+                    Err(Error::Platform {
+                        code: e.code().0 as i64,
+                        message: "Invoke failed".to_string(),
+                    })
+                }
+            })?;
             return Ok(());
         }
         // Try TogglePattern (checkboxes, switches)
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationTogglePattern>(UIA_TogglePatternId)
         } {
-            unsafe { pattern.Toggle() }
-                .or_else(|e| {
-                    if is_event_subscriber_failure(&e) {
-                        Ok(())
-                    } else {
-                        Err(Error::Platform {
-                            code: e.code().0 as i64,
-                            message: "Toggle failed".to_string(),
-                        })
-                    }
-                })?;
+            unsafe { pattern.Toggle() }.or_else(|e| {
+                if is_event_subscriber_failure(&e) {
+                    Ok(())
+                } else {
+                    Err(Error::Platform {
+                        code: e.code().0 as i64,
+                        message: "Toggle failed".to_string(),
+                    })
+                }
+            })?;
             return Ok(());
         }
         // Try SelectionItemPattern (list items, radio buttons)
@@ -778,17 +776,16 @@ impl Provider for WindowsProvider {
                 UIA_SelectionItemPatternId,
             )
         } {
-            unsafe { pattern.Select() }
-                .or_else(|e| {
-                    if is_event_subscriber_failure(&e) {
-                        Ok(())
-                    } else {
-                        Err(Error::Platform {
-                            code: e.code().0 as i64,
-                            message: "Select failed".to_string(),
-                        })
-                    }
-                })?;
+            unsafe { pattern.Select() }.or_else(|e| {
+                if is_event_subscriber_failure(&e) {
+                    Ok(())
+                } else {
+                    Err(Error::Platform {
+                        code: e.code().0 as i64,
+                        message: "Select failed".to_string(),
+                    })
+                }
+            })?;
             return Ok(());
         }
         // Try ExpandCollapsePattern (combo boxes, tree items)
@@ -851,17 +848,16 @@ impl Provider for WindowsProvider {
         if let Ok(pattern) = unsafe {
             uia_element.GetCurrentPatternAs::<IUIAutomationTogglePattern>(UIA_TogglePatternId)
         } {
-            unsafe { pattern.Toggle() }
-                .or_else(|e| {
-                    if is_event_subscriber_failure(&e) {
-                        Ok(())
-                    } else {
-                        Err(Error::Platform {
-                            code: e.code().0 as i64,
-                            message: "Toggle failed".to_string(),
-                        })
-                    }
-                })?;
+            unsafe { pattern.Toggle() }.or_else(|e| {
+                if is_event_subscriber_failure(&e) {
+                    Ok(())
+                } else {
+                    Err(Error::Platform {
+                        code: e.code().0 as i64,
+                        message: "Toggle failed".to_string(),
+                    })
+                }
+            })?;
             return Ok(());
         }
         Err(Error::ActionNotSupported {
@@ -877,17 +873,16 @@ impl Provider for WindowsProvider {
                 UIA_SelectionItemPatternId,
             )
         } {
-            unsafe { pattern.Select() }
-                .or_else(|e| {
-                    if is_event_subscriber_failure(&e) {
-                        Ok(())
-                    } else {
-                        Err(Error::Platform {
-                            code: e.code().0 as i64,
-                            message: "Select failed".to_string(),
-                        })
-                    }
-                })?;
+            unsafe { pattern.Select() }.or_else(|e| {
+                if is_event_subscriber_failure(&e) {
+                    Ok(())
+                } else {
+                    Err(Error::Platform {
+                        code: e.code().0 as i64,
+                        message: "Select failed".to_string(),
+                    })
+                }
+            })?;
             return Ok(());
         }
         Err(Error::ActionNotSupported {

